### PR TITLE
Add JSDoc comments for media state management

### PR DESCRIPTION
**Pull request by cvolzke4**
## Summary

Add JSDoc comments for media state management.

### CODE REVIEW

**Review by ezxyz — APPROVED** [review]

**Comment by swissspidy** [comment]
@cvolzke4 You might be interested in #2960 - trying to get to better JSDoc step by step

File: assets/src/edit-story/app/media/media3p/useContextValueProvider.js [hunk 1]
```diff
@@ -38,6 +38,13 @@ function useProviderSetContextValueProvider(reducerState, reducerActions) {
   return result;
 }
 
+/**
+ * Context fragment provider for 3rd party media (Unsplash, Coverr, etc).
+ * This is called from {@link MediaProvider} to provide the media global state.
+ *
+ * @param reducerState
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/media3p/useContextValueProvider.js]: Worth adding types and descriptions here everywhere too?

File: assets/src/edit-story/app/media/common/reducer.js [hunk 2]
```diff
@@ -33,6 +33,14 @@ export const INITIAL_STATE = {
   isMediaLoaded: false,
 };
 
+/**
+ * The reducer for the state of a media list pagination.
+ *
+ * This is called by the reducers for the state nodes:
+ * media/local, media/media3p/unsplash, media/media3p/coverr, etc.
+ *
+ * @param state
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/common/reducer.js]: Worth adding types, descriptions, defaults here too? (same for other files)

File: assets/src/edit-story/app/media/media3p/useContextValueProvider.js (hunk 1 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/media3p/useContextValueProvider.js]: Done.

File: assets/src/edit-story/app/media/common/reducer.js (hunk 2 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/common/reducer.js]: Done.

**Comment by cvolzke4** [comment]
> @cvolzke4 You might be interested in #2960 - trying to get to better JSDoc step by step

This is a great improvement, thanks for adding it!

**Pull request merged**